client/styles/ui-components/Chakra-Button.tsx

```diff
@@ -4,137 +4,153 @@ type ChakraButtonProps = {
   children?: React.ReactNode;
   onClick?: (event: React.MouseEvent) => void;
   as?: React.ElementType;
-}
+};
 
-export const ButtonBlack = ({children, as, onClick}:ChakraButtonProps) => {
-  return <Button
-  size="sm"
-  bgColor= "#000000"
-  _hover={{cursor: "pointer", bg: "#30363D"}}
-  textColor= 'white'
-  m="5px"
-  fontSize="14px"
-  onClick={onClick}
-  as={as}
-  >
-  {children}
-  </Button>
-}
+export const ButtonBlack = ({ children, as, onClick }: ChakraButtonProps) => {
+  return (
+    <Button
+      size="sm"
+      bgColor="#000000"
+      _hover={{ cursor: 'pointer', bg: '#30363D' }}
+      textColor="white"
+      m="5px"
+      fontSize="14px"
+      onClick={onClick}
+      as={as}
+    >
+      {children}
+    </Button>
+  );
+};
 
-export const ButtonGray = ({children, as, onClick}:ChakraButtonProps) => {
-  return <Button
-    bgColor="#EDF2F7"
-    textColor="black"
-    _hover={{cursor: "pointer", bg: "#E2E9F0"}}
-    m="5px"
-    ms="0px"
-    w="auto"
-    fontSize="14px"
-    onClick={onClick}
-    as={as}
+export const ButtonGray = ({ children, as, onClick }: ChakraButtonProps) => {
+  return (
+    <Button
+      bgColor="#EDF2F7"
+      textColor="black"
+      _hover={{ cursor: 'pointer', bg: '#E2E9F0' }}
+      m="5px"
+      ms="0px"
+      w="auto"
+      fontSize="14px"
+      onClick={onClick}
+      as={as}
     >
       {children}
     </Button>
+  );
 };
 
-export const ButtonGray2 = ({children, as,  onClick}:ChakraButtonProps) => {
-  return <Button
-    size="sm"
-    bgColor="#E2E9F0"
-    textColor="black"
-    _hover={{cursor: "pointer", bg: "#EDF2F7"}}
-    m="5px"
-    ms="0px"
-    w="auto"
-    fontSize="14px"
-    onClick={onClick}
-    as={as}
+export const ButtonGray2 = ({ children, as, onClick }: ChakraButtonProps) => {
+  return (
+    <Button
+      size="sm"
+      bgColor="#E2E9F0"
+      textColor="black"
+      _hover={{ cursor: 'pointer', bg: '#EDF2F7' }}
+      m="5px"
+      ms="0px"
+      w="auto"
+      fontSize="14px"
+      onClick={onClick}
+      as={as}
     >
       {children}
     </Button>
+  );
 };
 
-export const ButtonGreen = ({children,as, onClick}:ChakraButtonProps) => {
-  return <Button
-    size="sm"
-    bgColor="#38A169"
-    textColor="white"
-    _hover={{cursor: "pointer", bg: "#2F855A"}}
-    m="5px"
-    ms="0px"
-    w="auto"
-    fontSize="14px"
-    onClick={onClick}
-    as={as}
+export const ButtonGreen = ({ children, as, onClick }: ChakraButtonProps) => {
+  return (
+    <Button
+      size="sm"
+      bgColor="#38A169"
+      textColor="white"
+      _hover={{ cursor: 'pointer', bg: '#2F855A' }}
+      m="5px"
+      ms="0px"
+      w="auto"
+      fontSize="14px"
+      onClick={onClick}
+      as={as}
     >
       {children}
     </Button>
+  );
 };
```

Comment:
I'm seeing a lot of repetition in this file. Most of these variants of the Chakra `Button` component share most of their props. Can we clean this up?

Reply:
True, I don't remember changing this page it might just be style changes that show up in the git log, I'll look into it either way 😃

Reply:
> 

These buttons will probably be changed, but you're right about cleaning the code